src/vaults/AsyncRequestManager.sol

```diff
@@ -151,7 +168,21 @@ contract AsyncRequestManager is Auth, Recoverable, IAsyncRequestManager {
     }
 
     function _sendRequest(IBaseVault vault_, bytes memory payload) internal {
-        spoke.request(vault_.poolId(), vault_.scId(), spoke.vaultDetails(vault_).assetId, payload);
+        IRefundEscrow refund;
+        uint256 payment;
+
+        if (!balanceSheet.gateway().isBatching()) {
+            refund = refundEscrowFactory.get(vault_.poolId());
+            require(address(refund).code.length > 0, RefundEscrowNotDeployed());
+
+            refund.withdrawFunds(address(this), address(refund).balance); // All funds goes to this contract
+            payment = address(this).balance;
```

Comment:
Curious if it can ever happen that `AsyncRequestManager` has residual ETH from other source. Then, the pool subsidy would use those funds as well. In case of that, we should rather do the following, though I suppose my question is purely theoretical

```suggestion
            uint256 availableSubsidy = address(refund).balance;
            refund.withdrawFunds(address(this), availableSubsidy);
            payment = availableSubsidy;
```

Reply:
Good theoretical question. Should not have any eth, but I agree this could be cleaner.